File: backend/entity/Verify.go
package entity

import (
	"time"

	"gorm.io/gorm"
)

type Verify struct {
	gorm.Model
	VerificationDocument string     `json:"verification_document"`
	Reason               string     `json:"reason"` // ใช้กรอกเหตุผลเมื่อสถานะเป็น "ปฏิเสธ"
	VerifiedAt           *time.Time `json:"verified_at"`
	
	StatusVerifyID       uint
	StatusVerify         StatusVerify `gorm:"foreignKey:StatusVerifyID"`

	UserID uint
	User   User `gorm:"foreignKey:UserID"`

	AdminID *uint  // nullable เพราะตอนอัปโหลดใหม่ ยังไม่มี admin
	Admin   *Admin `gorm:"foreignKey:AdminID"`
}

File: frontend/src/interfaces/Verify.ts
import type { StatusVerifyInterface } from "./StatusVerify";
import type { UserInterface } from "./User";
import type { AdminInterface } from "./Admin";

export interface VerifyInterface {
  ID?: number;
  verification_document?: string;
  reason?: string;
  verified_at?: string;

  StatusVerifyID?: number;
  StatusVerify?: StatusVerifyInterface;

  UserID?: number;
  User?: UserInterface;

  AdminID?: number | null;
  Admin?: AdminInterface | null;

  CreatedAt?: string;
  DeletedAt?: string | null;
  UpdatedAt?: string;
}

export interface VerifyStatsResponse {
  total: number;
  not_submitted: number;
  pending: number;
  approved: number;
  rejected: number;
  by_status: { status: string; count: number }[];
}